resources/party.ts

```diff
@@ -41,99 +52,99 @@ export default class PartyTracker {
 
   // returns an array of the names of tanks in your immediate party
-  get tankNames() {
+  get tankNames(): string[] {
     return this.roleToPartyNames_['tank'];
   }
 
   // returns an array of the names of healers in your immediate party
-  get healerNames() {
+  get healerNames(): string[] {
     return this.roleToPartyNames_['healer'];
   }
 
   // returns an array of the names of dps players in your immediate party
-  get dpsNames() {
+  get dpsNames(): string[] {
     return this.roleToPartyNames_['dps'];
   }
 
   // returns true iff the named player in your alliance is a particular role
-  isRole(name, role) {
+  isRole(name: string, role: string): boolean {
     return this.nameToRole_[name] === role;
   }
 
   // returns true iff the named player in your alliance is a tank
-  isTank(name) {
+  isTank(name: string): boolean {
     return this.isRole(name, 'tank');
   }
 
   // returns true iff the named player in your alliance is a healer
-  isHealer(name) {
+  isHealer(name: string): boolean {
     return this.isRole(name, 'healer');
   }
 
   // returns true iff the named player in your alliance is a dps
-  isDPS(name) {
+  isDPS(name: string): boolean {
     return this.isRole(name, 'dps');
   }
 
   // returns true iff the named player is in your immediate party
-  inParty(name) {
+  inParty(name: string): boolean {
     return this.partyNames.includes(name);
   }
 
   // returns true iff the named player is in your alliance
-  inAlliance(name) {
+  inAlliance(name: string): boolean {
     return this.allianceNames.includes(name);
   }
 
   // for a named player, returns the other tank in your immediate party
   // if named player is not a tank, or there's not exactly two tanks
   // in your immediate party, returns null.
-  otherTank(name) {
+  otherTank(name: string): string | null {
     const names = this.tankNames;
     if (names.length !== 2)
       return null;
     if (names[0] === name)
-      return names[1];
+      return names[1] as string;
     if (names[1] === name)
-      return names[0];
+      return names[0] as string;
     return null;
   }
 
   // see: otherTank, but for healers.
-  otherHealer(name) {
-    const names = this.roleToPartyNames['healer'];
+  otherHealer(name: string): string | null {
+    const names = this.roleToPartyNames_['healer'];
     if (names.length !== 2)
       return null;
     if (names[0] === name)
-      return names[1];
+      return names[1] as string;
     if (names[1] === name)
-      return names[0];
+      return names[0] as string;
     return null;
   }
 
   // returns the job name of the specified party member
-  jobName(name) {
+  jobName(name: string): Job | null {
     const partyIndex = this.partyNames.indexOf(name);
     if (partyIndex >= 0)
-      return Util.jobEnumToJob(this.details[partyIndex].job);
+      return Util.jobEnumToJob(this.details[partyIndex]?.job as number);
     return null;
   }
 
-  nameFromId(id) {
+  nameFromId(id: string): string | undefined {
```

Comment:
Maybe this is for a separate PR, but it is definitely odd that this one returns undefined but the others return null.

On the other hand, I think having the rest return undefined instead of null would clean up a bunch of the types.  You could say stuff like this, without needing all of the `as string` lines:

```
  otherHealer(name: string): string | undefined {
    const names = this.roleToPartyNames_['healer'];
    if (names.length !== 2)
      return;
    if (names[0] === name)
      return names[1];
    if (names[1] === name)
      return names[0];
  }
```

What do you think?

Reply:
So you means to modify other methods to return `undefined` too?

Reply:
Yeah, I think I was suggesting that they all could return `string | undefined` and wondering if that would be more consistent / clean things up.  Feel free to disagree here.  It was more just me thinking out loud and wondering about the `as string` and the inconsistency.